sdk/src/main/java/com/hedera/hashgraph/sdk/Network.java

```diff
@@ -207,9 +211,14 @@ private static Map<AccountId, NodeAddress> getAddressBookForLedger(@Nullable Led
 
     static HashMap<String, AccountId> addressBookToNetwork(Collection<NodeAddress> addressBook, int desiredPort) {
         var network = new HashMap<String, AccountId>();
+        boolean wellKnownPorts = addressBook.stream()
+            .map(NodeAddress::getAddresses)
+            .flatMap(Collection::stream)
+            .map(Endpoint::getPort)
+            .allMatch(Set.of(PORT_NODE_TLS, PORT_NODE_PLAIN)::contains);
         for (var nodeAddress : addressBook) {
             for (var endpoint : nodeAddress.addresses) {
-                if (endpoint.port == desiredPort) {
+                if (!wellKnownPorts || endpoint.port == desiredPort) {
```

Comment:
I guess we could remove `int desiredPort` and remove this if statement as a whole. WDYT?

Reply:
It's ok with me, but it changes the behavior of things. Currently this API only allows a single port through, if you remove desiredPort it will allow any and all ports through.

Also, this change should be made in conjunction with the change to BaseNodeAddress. Because this change will allow non-standard port numbers to get through, but without the corresponding change in BaseNodeAddress requests to use transport security on non-standard ports will be silently ignored.

Reply:
`desiredPort` right now can be either 50212 or 50211 and given this information, the current filtering does not make sense for me. 
If we want to allow not well known ports, it makes sense for me to allow any port.
At the end of the day, the user can set whatever network they want given the addition of dynamic address book feature.